src/components/address/address.service.ts

```diff
@@ -0,0 +1,93 @@
+import { HttpException, HttpStatus, Injectable, Logger } from '@nestjs/common';
+import { InjectRepository } from '@nestjs/typeorm';
+import { Repository } from 'typeorm';
+
+import { Address } from '../../database/entities/address.entity';
+import { CreateAddressDto } from './dto/createAddress.dto';
+import { UpdateAddressDto } from './dto/updateAddress.dto';
+import { CRUD } from 'src/utils/CRUD.interface';
+import { User } from '../../database/entities/user.entity';
+
+@Injectable()
+export class AddressService implements CRUD<Address> {
+  private readonly logger = new Logger('AddressService');
+
+  constructor(
+    @InjectRepository(Address)
+    private readonly addressRepo: Repository<Address>,
+  ) {}
+
+  async create(address: CreateAddressDto, user: User) {
+    try {
+      return await this.addressRepo.save(address);
+    } catch (error) {
+      this.logger.error(error);
+      throw new HttpException(
```

Comment:
por qué usar un HttpException en lugar de lanzar uno de los errors que ya da nestjs? por qué el no poder crear un recurso es un BAD_GATEWAY en lugar de un InternalServerError?

Reply:
estan manejando de forma incorrecta los errores, la forma en que trabaja Nestjs es con https://docs.nestjs.com/exception-filters#built-in-http-exceptions, decidan si van a seguir como lo tienes o si usan herramientas que da el framework